### ISSUE TITLE
AxClient generator crashes if `n_init < sim_workers`

### ISSUE THREAD

**Issue by AngelFP** [optimas-org/optimas#133]
We should handle this better, so that if `n_init` is small, the workers >=5 should be idle until the first random batch finishes. Alternatively, set `n_init=sim_workers`.

**Comment by n01r** [comment]
Ah, I just ran into this as well when changing the number of workers.
I would have assumed the initial `n_init` simulations would be used to build the surrogate, and then the `sim_workers` would choose their parameter values.

But indeed, the error message from the crash could be clearer. Currently, it is something like:

```
[0]  2024-01-10 16:23:23,377 libensemble.manager (ERROR): Message: ax.exceptions.core.DataRequiredError: `StandardizeY` transform requires non-empty data.
[0]  2024-01-10 16:23:23,378 libensemble.manager (ERROR): Traceback (most recent call last):
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/libensemble/worker.py", line 406, in run
    response = self._handle(Work)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/libensemble/worker.py", line 352, in _handle
    calc_out, persis_info, calc_status = self._handle_calc(Work, calc_in)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/libensemble/worker.py", line 273, in _handle_calc
    out = calc(calc_in, Work)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/libensemble/utils/runners.py", line 62, in run_gen
    return result(calc_in, Work["persis_info"], self.gen_specs, Work["libE_info"], self.gen_f, Work["tag"])
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/libensemble/utils/runners.py", line 95, in _normal_result
    return user_f(*args)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/optimas/gen_functions.py", line 70, in persistent_generator
    generated_trials = generator.ask(1)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/optimas/generators/base.py", line 184, in ask
    gen_trials = self._ask(gen_trials)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/optimas/generators/ax/service/base.py", line 90, in _ask
    parameters, trial_id = self._ax_client.get_next_trial()
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/utils/common/executils.py", line 161, in actual_wrapper
    return func(*args, **kwargs)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/service/ax_client.py", line 549, in get_next_trial
    generator_run=self._gen_new_generator_run(), ttl_seconds=ttl_seconds
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/service/ax_client.py", line 1771, in _gen_new_generator_run
    return not_none(self.generation_strategy).gen(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/generation_strategy.py", line 281, in gen
    return self._gen_multiple(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/generation_strategy.py", line 399, in _gen_multiple
    self._fit_current_model(data=data)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/generation_strategy.py", line 468, in _fit_current_model
    self._curr.fit(experiment=self.experiment, data=data, **model_state_on_lgr)
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/generation_node.py", line 243, in fit
    model_spec.fit(  # Stores the fitted model as `model_spec._fitted_model`
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/model_spec.py", line 145, in fit
    self._fitted_model = self.model_enum(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/registry.py", line 386, in __call__
    model_bridge = bridge_class(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/torch.py", line 129, in __init__
    super().__init__(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/base.py", line 210, in __init__
    observations, search_space = self._transform_data(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/base.py", line 297, in _transform_data
    t_instance = t(
  File "/global/cfs/cdirs/m3239/mgarten/sw/perlmutter/conda_envs/wake-t_env/lib/python3.10/site-packages/ax/modelbridge/transforms/standardize_y.py", line 45, in __init__
    raise DataRequiredError("`StandardizeY` transform requires non-empty data.")
ax.exceptions.core.DataRequiredError: `StandardizeY` transform requires non-empty data.
```

**Comment by AngelFP** [comment]
Hi @n01r,
Glad to see the issue was of help ;)
The first `n_init` evaluations are generated without using a surrogate (SOBOL sampling is used instead), but they are still taken into account when building the surrogate later. The issue here is that if, for example,  `n_init=2` and `sim_workers=4` , the first two workers will run the SOBOL trials, while the other two will request trials made using BO (i.e., generated from a surrogate that cannot be built, because there is no data available yet).

**Comment by AngelFP** [comment]
Solved in #207
